src/libopensc/card-openpgp.c

```diff
@@ -629,19 +731,36 @@ pgp_get_card_features(sc_card_t *card)
 			unsigned long flags;
 
 			/* Is this correct? */
-			/* OpenPGP card spec 1.1 & 2.0, section 7.2.9 & 7.2.10 */
+			/* OpenPGP card spec 1.1 & 2.x, section 7.2.9 & 7.2.10 / v3.x section 7.2.11 & 7.2.12 */
 			flags = SC_ALGORITHM_RSA_PAD_PKCS1;
 			flags |= SC_ALGORITHM_RSA_HASH_NONE;
 			/* Can be generated in card */
 			flags |= SC_ALGORITHM_ONBOARD_KEY_GEN;
 
-			if ((pgp_get_blob(card, blob73, i, &blob) >= 0) &&
-				(blob->data != NULL) && (blob->len >= 4)) {
-				if (blob->data[0] == 0x01) {	/* Algorithm ID [RFC4880]: RSA */
-					unsigned int keylen = bebytes2ushort(blob->data + 1);  /* Measured in bit */
+			/* OpenPGP card spec 1.1 & 2.x section 4.3.3.6 / v3.x section 4.4.3.7 */
+			if ((pgp_get_blob(card, blob73, i, &blob) >= 0) && (blob->data != NULL)) {
+				if (blob->len >= 3 && blob->data[0] == 0x01) {	/* RSA [RFC 4880] */
+					unsigned int keybits = bebytes2ushort(blob->data + 1);
 
-					_sc_card_add_rsa_alg(card, keylen, flags, 0);
+					_sc_card_add_rsa_alg(card, keybits, flags, 0);
 				}
+				/* v3.0+: [RFC 4880 & 6637] 0x12 = ECDH, 0x13 = ECDSA */
+			}
+		}
+
+		if (priv->bcd_version >= OPENPGP_CARD_3_0) {
+			/* v3.0+: get length info from "extended length information" DO */
+			if ((pgp_get_blob(card, blob6e, 0x7f66, &blob) >= 0) &&
+				(blob->data != NULL) && (blob->len >= 8)) {
+				/* kludge: treat as SIMPLE DO and use appropriate offsets */
+				card->max_send_size = bebytes2ushort(blob->data + 2);
+				card->max_recv_size = bebytes2ushort(blob->data + 6);
+			}
+
+			/* v3.0+: get card features from "general feature management" DO */
+			if ((pgp_get_blob(card, blob6e, 0x7f74, &blob) >= 0) &&
```

Comment:
Why read that data if you don't need it?

The same is true for other parsed values or data structures, e.g. `max_specialDO_size`.

I'd better like to see that included when we're actually doing something useful with it. Currently, that's dead weight!

Reply:
OK, understandable.

Instead of amending this PR and reverting the chanes in additional commits, I lean to calceling this one, and putting in a new one with the offending passages removed.

What's your preference?

Reply:
choose as you like, thanks